### PR TITLE
Week2 nadin

### PR DESCRIPTION
**Pull request by Nadoshka**

### CODE REVIEW

File: Week2/part1.js [hunk 1]
```diff
@@ -0,0 +1,81 @@
...
+    }
+  });
+}
+
+const inputType = process.argv[2]; // type like region, language, country
+const userInput = process.argv[3]; // value like western europe, Dutch, Netherlands
+const userInput2 = process.argv[4];
+
+//What is the capital of country X ? (Accept X from user)
+const capital_of_country = `SELECT capital FROM country WHERE country.Name = '${userInput}'`;
```
Comment (gajduk) [inline Week2/part1.js]: What would happen if the user with the query
`Spain; drop table country` ?
To protect your app against such attacks you need to use prepared statements. See an example here https://github.com/mysqljs/mysql#preparing-queries

File: Week2/part1.js [hunk 2]
```diff
@@ -0,0 +1,81 @@
...
+function help() {
+  console.log(`
+  - to find the capital of country X enter 'country' followed by the name of the country you want.
+  - to List all the languages spoken in the region Y enter 'region' followed by the region you want.
+  - to find the number of cities in which language Z is spoken enter 'language' followed by the language you want.
+`);
+}
+
+//Find the number of cities in which language Z is spoken (Accept Z from user)
+const language_of_city = `SELECT COUNT(Name) AS CityLanguage FROM city INNER JOIN countrylanguage ON countrylanguage.CountryCode = city.Country_Code WHERE countrylanguage.Language = '${userInput}'`;
```
Comment (gajduk) [inline Week2/part1.js]: Nice

File: Week2/part1.js [hunk 3]
```diff
@@ -0,0 +1,81 @@
...
+
+//Accept the region and language from the user. Are there any countries in this region with the given language as the official language ? If yes, display those countries. If no, display FALSE. E.g. (A) input region : 'Western Europe' and input language : 'Dutch' output should be Belgium and Netherlands (B) input region : 'Western Europe' and input language : 'Hindi' output should be 'FALSE'
+
+const official_language = `SELECT Name FROM country INNER JOIN countrylanguage ON countrylanguage.CountryCode = country.Code WHERE countrylanguage.Language = '${userInput}' AND country.Region = '${userInput2}' AND countrylanguage.isOfficial ='T'`;
+
+// List all the continents with the number of languages spoken in each continent
+const language_of_continents =
+  'SELECT COUNT(countrylanguage.Language) as languages, country.continent FROM country INNER JOIN countrylanguage ON countrylanguage.CountryCode = country.Code GROUP BY country.continent';
```
Comment (gajduk) [inline Week2/part1.js]: Suppose there is a small continent with two countries that speak the same language. On this continent only 1 language is spoken. However your query would return a wrong result. How can this be fixed?

File: Week2/part2.js [hunk 4]
```diff
@@ -0,0 +1,54 @@
...
+CREATE TRIGGER my_trigger 
+    BEFORE INSERT
+        ON countrylanguage
+            FOR EACH ROW
+            BEGIN
+                DECLARE message VARCHAR(100);
+                DECLARE sd INT;
+                SET sd= (SELECT COUNT(countrylanguage.Language) AS Languages
+                FROM countrylanguage
+                WHERE countrylanguage.CountryCode = CountryCode);
```
Comment (gajduk) [inline Week2/part2.js]: I am not sure I understand where is the second CountryCode comming from?

File: Week2/part1.js (hunk 1 above)
Comment (Nadoshka) [reply Week2/part1.js]: It's done! Thanks for the information! I appreciate it! :)

File: Week2/part2.js (hunk 4 above)
Comment (Nadoshka) [reply Week2/part2.js]: It's from the country table. "country.CountryCode" 
It's done! Thanks for your feedback! :)

**Comment by gajduk** [comment]
Hi @Nadoshka  can you implement the changes that we discussed on Sunday. Then I can give positive feedback on the homework. Thank you!

**Pull request closed without merging**